### PR TITLE
feat(smn): Add Resource huaweicloud_smn_logtank

### PR DESCRIPTION
**Pull request by dangyy0626**
**What this PR does / why we need it**:
add resource huaweicloud_smn_logtank provider

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed
```
 make testacc TEST="./huaweicloud/services/acceptance/smn" TESTARGS="-run TestAccSMNV2Logtank_basic"    
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/smn -v -run TestAccSMNV2Logtank_basic -timeout 360m -parallel 4 
=== RUN   TestAccSMNV2Logtank_basic 
=== PAUSE TestAccSMNV2Logtank_basic
=== CONT  TestAccSMNV2Logtank_basic
--- PASS: TestAccSMNV2Logtank_basic (52.94s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/smn       52.969s
```

### CODE REVIEW

File: docs/resources/smn_logtank.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+* `logtank_id` - The resource ID.
+
+* `created_at` - Time when the logtank was created.
+
+* `updated_at` - Time when the logtank was updated.
+
+## Import
+
+SMN topic can be imported using the `topic_urn` or using the `topic_urn` / `logtank_id`.
```
Comment (Jason-Zhang9309) [inline docs/resources/smn_logtank.md]:
```suggestion
SMN topic can be imported using the `topic_urn` or using the `topic_urn` and `logtank_id` separated by a slash e.g.
```

File: docs/resources/smn_logtank.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the SMN topic resource. If omitted, the
+  provider-level region will be used. Changing this parameter will create a new resource.
+
+* `topic_urn` - (Required, String, ForceNew) Resource identifier of a topic, which is unique.
```
Comment (Jason-Zhang9309) [inline docs/resources/smn_logtank.md:42]:
```suggestion
* `topic_urn` - (Required, String, ForceNew) Resource identifier of a topic, which is unique.
  Changing this parameter will create a new resource.
```

File: docs/resources/smn_logtank.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+* `log_group_id` - (Required, String) The lts log group ID.
+
+* `log_stream_id` - (Required, String) The lts log stream ID.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
```
Comment (Jason-Zhang9309) [inline docs/resources/smn_logtank.md:51]: * `id` - The resource ID. The value is the topic URN.

File: docs/resources/smn_logtank.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+* `logtank_id` - The resource ID.
```
Comment (Jason-Zhang9309) [inline docs/resources/smn_logtank.md]:
```suggestion
* `id` - The resource ID. The value is the topic URN.

* `logtank_id` - The ID of the logtank.
```

File: huaweicloud/services/smn/resource_huaweicloud_smn_logtank.go [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+		return diag.Errorf("error setting logtankGet: %s", err)
+	}
+	return nil
+}
+
+func getLogtankById(logtanks []logtank.LogtankGet, id string) *logtank.LogtankGet {
+	if len(logtanks) == 0 {
+		return nil
+	}
+	if len(id) == 0 {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/smn/resource_huaweicloud_smn_logtank.go:163]:
```suggestion
	if id == "" {
```

Comment (Jason-Zhang9309) [inline huaweicloud/services/smn/resource_huaweicloud_smn_logtank.go]:
```suggestion
	if id == "" {
```

File: huaweicloud/services/acceptance/smn/resource_huaweicloud_smn_logtank_test.go [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func getResourceSMNLogtankFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := cfg.SmnV2Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating smn v2 client: %s", err)
+	}
+
+	return logtank.List(client, state.Primary.ID).Extract()
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/smn/resource_huaweicloud_smn_logtank_test.go]: getLogtankById should also be used here

File: huaweicloud/services/smn/resource_huaweicloud_smn_logtank.go [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+	error) {
+	var mErr *multierror.Error
+	parts := strings.Split(d.Id(), "/")
+	d.SetId(parts[0])
+	if len(parts) == 2 {
+		mErr = multierror.Append(d.Set("logtank_id", parts[1]))
+	} else if len(parts) > 2 {
+		return nil, fmt.Errorf("the imported ID specifies an invalid format, must be <topic_urn> or <topic_urn>/<id>")
+	}
+	return []*schema.ResourceData{d}, mErr.ErrorOrNil()
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/smn/resource_huaweicloud_smn_logtank.go]:
```suggestion
	parts := strings.Split(d.Id(), "/")
	if len(parts) < 1 || len(parts) > 2  {
		return nil, fmt.Errorf("the imported ID specifies an invalid format, must be <topic_urn> or <topic_urn>/<id>")
        }
	d.SetId(parts[0])
	if len(parts) == 2 {
		mErr = multierror.Append(d.Set("logtank_id", parts[1]))
	}
	
	return []*schema.ResourceData{d}, mErr.ErrorOrNil()
```

File: huaweicloud/services/smn/resource_huaweicloud_smn_logtank.go [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/smn/v2/logtank"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+const TopicNotExistsCode = "SMN.00010008"
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/smn/resource_huaweicloud_smn_logtank.go]:
```suggestion
const LogtankNotExistsCode = "SMN.00010008"
```

File: huaweicloud/services/acceptance/smn/resource_huaweicloud_smn_logtank_test.go [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+			},
+		},
+	})
+}
+
+func testAccSMNV2ImportStateIdFunc(logtankResourceName string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		logtank, ok := s.RootModule().Resources[logtankResourceName]
+		if !ok {
+			return "", fmt.Errorf("Auto smn logtank not found: %s", logtank)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/smn/resource_huaweicloud_smn_logtank_test.go]:
```suggestion
			return "", fmt.Errorf("smn logtank not found: %s", logtank)
```

File: docs/resources/smn_logtank.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  log_group_id  = huaweicloud_lts_group.lts_group_test.id
+  log_stream_id = huaweicloud_lts_stream.lts_stream_test.id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the SMN topic resource. If omitted, the
```
Comment (ShiChangkuo) [inline docs/resources/smn_logtank.md]: the resource is SMN logtank not topic

**Pull request merged**